src/plugins/discover/public/application/main/hooks/use_esql_mode.test.tsx

```diff
@@ -150,10 +150,11 @@ describe('useEsqlMode', () => {
     });
   });
 
-  test('changing an ES|QL query with same result columns should not change state when loading and finished', async () => {
+  test('changing an ES|QL query with same result columns should change state when loading and finished', async () => {
```

Comment:
```suggestion
  test('changing an ES|QL query with same result columns but a different index pattern should change state when loading and finished', async () => {
```
Maybe this makes it more clearer